src/run/run_step.go

```diff
@@ -55,22 +61,30 @@ func Parallel(ctx context.Context, state *core.BuildState, labels []core.Annotat
 // Sequential runs a series of targets sequentially.
 // Returns a relevant exit code (i.e. if at least one subprocess exited unsuccessfully, it will be
 // that code, otherwise 0 if all were successful).
-func Sequential(state *core.BuildState, labels []core.AnnotatedOutputLabel, args []string, quiet, remote, env bool, dir string) int {
+func Sequential(state *core.BuildState, labels []core.AnnotatedOutputLabel, args []string, quiet, remote, env, inTmp bool, dir string) int {
+	prepareRun(dir, inTmp)
 	for _, label := range labels {
 		log.Notice("Running %s", label)
-		if err := run(state, label, args, true, quiet, remote, env, false, dir); err != nil {
+		if err := run(state, label, args, true, quiet, remote, env, false, inTmp, dir); err != nil {
 			log.Error("%s", err)
 			return err.(*exitError).code
 		}
 	}
 	return 0
 }
 
+func prepareRun(dir string, inTmp bool) {
+	if inTmp && dir != "" {
+		log.Fatal("Can't use both --in_temp_dir and --in_wd at the same time")
```

Comment:
This should happen in `please.go` - we shouldn't need to refer to flags here.
(Also mildly nicer to error out immediately rather than do all the building then fail...)